### DISCUSSION TITLE
Passing arguments to the call_capl_function - invalid len counting

### DISCUSSION THREAD

**Discussion by jaran94** [chaitu-ycr/py_canoe#52] [Q&A]
Hi, 
I've been struggling with the passing the tuple with the arguments into the call_capl_function method. When I'm passing the tuple with arguments, for example (1, 2, 3, 4) and when these get passed to the capl_obj.call_capl_function they are treated like a single argument (single tuple) and therefore counted as 1.

 I've tested the solution when I'm switching to arguments = arguments[0] and then this check len(arguments) == self.parameter_count(capl_function_obj)  is True, but error appears:   
File "C:\Users\xxxx\AppData\Local\Temp\gen_py\3.9\7F31DEB0-5BCC-11D3-8562-00105A3E017Bx0x1x70.py", line 3440, in Call
    return self._ApplyTypes_(3, 1, (12, 0), ((12, 17), (12, 17), (12, 17), (12, 17), (12, 17), (12, 17), (12, 17), (12, 17), (12, 17), (12, 17)), 'Call', None,p1
  File "E:\xxx\xxx\xxx\venv\lib\site-packages\win32com\client\__init__.py", line 574, in _ApplyTypes_
    self._oleobj_.InvokeTypes(dispid, 0, wFlags, retType, argTypes, *args),
pywintypes.com_error: (-2147352567, 'Exception occurred.', (0, None, None, None, 0, -2147352571), 10)

And I don't know how to deal with that thing. Maybe I'm doing something wrong or maybe there's some workaround?

**Comment by chaitu-ycr** [comment]
hi @jaran94 

can you share snippet of your python code here.
ideally you dont have to pass tuple as argument.
check below example
![image](https://github.com/chaitu-ycr/py_canoe/assets/67551335/66f9fd10-932b-40ff-8c0f-7806650c63d5)

let me know your feedback
